arangod/GeneralServer/VstCommTask.cpp

```diff
@@ -134,6 +135,45 @@ bool VstCommTask<T>::readCallback(asio_ns::error_code ec) {
   return true; // continue read loop
 }
 
+template <SocketType T>
+void VstCommTask<T>::setIOTimeout() {
+  double secs = GeneralServerFeature::keepAliveTimeout();
+  if (secs <= 0) {
+    return;
+  }
+  
+  const bool wasReading = this->_reading;
+  const bool wasWriting = this->_writing;
+  TRI_ASSERT(wasReading || wasWriting);
+  if (wasWriting) {
+    secs = std::max(300.0, secs);
```

Comment:
why is the timeout at least 300 seconds hee? the 300 seconds minimum seems to be in place for HTTP/2 and VST, but not for HTTP/1. I don't see a reason for this.

Reply:
because i am not gonna change the http 1.1 behaviour

Reply:
Ok, but then this still leaves us with protocol-specific behavior:
* HTTP/1.1 doesn't have a write timeout
* HTTP/2 and VST have a write timeout which is max(idle timeout, 300)

Ideally we wouldn't have differences between the protocols, but try to establish the same behavior for all protocols. If that requires a write-timeout to be configurable, we should add it as a new configuration option.

Reply:
nah some people are sending GB image files as part of foxx apps and stuff like that (and we are not even using sendfile here). I'd rather keep it sane